### ISSUE TITLE
Add an element to all certificate pages

### ISSUE THREAD

**Issue by abias** [markn86/moodle-mod_customcert#183]
Currently, if an element is added to a certificate template, it is bound to the page it was added to. In some scenarios, we would like to be able to add an element to all certificate pages. This is to be able to reliably add headers, footers and logos to all pages of a multipage certificate.

**Comment by dmitriim** [comment]
@abias Looking at the plugin architecture, I see this as an additional element setting like 'Add to all pages". Once this setting is ON, the element will be "copied" to all existing pages. Basically it will appear in elements list for any page as a standalone element.
How would you see the logic and UI when you'd like to update or remove the element? Should we apply a change to an element with enabled "Add to all pages" to all copies of the element? Should we warn user somehow?

**Comment by abias** [comment]
@dmitriim: 
> Looking at the plugin architecture, I see this as an additional element setting like 'Add to all pages". Once this setting is ON, the element will be "copied" to all existing pages. Basically it will appear in elements list for any page as a standalone element.
Yes, this sounds like a feasible approach without rebuilding the whole plugin.

> How would you see the logic and UI when you'd like to update or remove the element? Should we apply a change to an element with enabled "Add to all pages" to all copies of the element? 
Yes, it would be great (and intuitive) if changes to / deletions of an element with "add to all pages" enabled would be automatically applied to the other instances of the element on all pages.

> Should we warn user somehow?
I don't think so as long as the user can recognize
1. from the element's detail page that he is editing an element which is placed on all pages
2. from the template overview page that he is deleting an element which is placed on all pages
3. from the template layouting page that he is editing an element which is placed on all pages
